### ISSUE TITLE
Project Feedback

### ISSUE THREAD

**Issue by by-the-w3i** [by-the-w3i/MI-449-SS17-741-js-serving-content-with-express-OctGcX#1]
@egillespie Can you take a look at this? It's [hosted here](https://jiangw14-kfxl35erbteu5octgcx.herokuapp.com/) and meets the following criteria:

- [x] No code is repeated in your files; create a partial for any shared elements (`nav`, `head`, `header`, etc.)
- [x] Make a navigation to move throughout your site, using `forEach` to create a set of links to other data pages
- [x] Each item should have an object with an image, a link, a title, and more properties based on your topic (i.e. sandwich ingredients, movie release date, etc.)
- [x] Include at least 3 data pages and a homepage; routes should be created in Express for each
- [x] Use data appropriately with HTML tags (paragraphs in `p` tags, headings in `h1`-`h6`, etc.)
- [x] Use `express.static` to create a folder for static assets (images, stylesheets, etc.)
- [x] Include a `.gitignore` file so that `node_modules` and any other unnecessary files are not commited to GitHub

**Comment by egillespie** [comment]
Hi @by-the-w3i, this looks really good! Well done with the templating. There's just one place where I think you could avoid repeating a bunch of code.

Instead of copying and pasting your route method each time you want to add a new URL:

```js
app.get('/', function (request, response) {
  response.render('pages/index', {
    items: items,
    item: items[0]
  })
})

app.get('/about', function (request, response) {
  response.render('pages/index', {
    items: items,
    item: items[1]
  })
})

// ...
```

You can use `forEach` to loop over `items` and create a route for each one:

```js
Object.keys(items).forEach(function (id) {
  var item = items[id]
  app.get(item.link, function (request, response) {
    response.render('pages/index', {
      items: items,
      item: item
    })
  })
})
```

Now you can add as many items as you want and you don't have to worry about creating routes for them.

Can you make this change and comment back when it's working?

Thanks! 🍍

**Comment by by-the-w3i** [comment]
@egillespie That's neat!!! I really like it!!! Thank you for pointing this out!! I have fixed it. Please take a look thank you very much! I appreciate it!!

**Comment by egillespie** [comment]
Yeah, looping is a big time saver and can be used in a lot of clever ways! Good work on incorporating that change into your code. :shipit:
